### ISSUE TITLE
Have Friend Requests show at the Top of the Contacts List

### ISSUE THREAD

**Issue by GearBell** [Neos-Metaverse/NeosPublic#2596]
For people with many many friends online its frustrating to have to scroll forever to find the friend requests that are at the bottom of the list. By putting requests at the top of the contacts list its easier to accept/deny on the spot, rather than avoiding the large amounts of scrolling and letting requests sit there. It would be much more user-friendly and easier for new users to understand too.

**Comment by TehTurk** [comment]
This wouldn't work too well, I actually recently spoke to @shiftyscales about this problem myself and it proves to be a problem. Some people use the contact requests as a form of like "Seeing if this person has been around long enough to justify adding them as a contact" For those with like 20-30 Contact requests, your friends list is effectively full all the time. Not going to close this, but I think this would be a good issue to note for the Contacts window redesign.

**Comment by Mentalish1** [comment]
But that's some but not all :o Side note to add in, even a function or tab to be able to differentiate between Friend Requests and Currently Active users online. Even a one off option in the settings that goes from "Friend Requests at the top" or "Friend Requests at the bottom".

**Comment by GearBell** [comment]
The request Im asking is not a permanent UI redisgn, just a change to have the requests be at the top, not the bottom. A hotfix, until future UI changes.

**Comment by Mysticporo** [comment]
As a Hot Fix i think this would be beneficial not only for myself but also others cosidering the majority of people have to scroll through their contacts list to begin to get near the request section,
While i can see that this could be decisive i believe the Hot Fix would be quickly accepted by many and useful for the user base.

**Comment by TehTurk** [comment]
Oh no your welcome to discuss sorry if it seemed like that, I had the same mindset when I brought up the issue. Just posting out some of the problems that kind come up with the issue. I don't really see it as a divisive issue but more of where like the life cycle of a request is usually Issue Made -> Discussed -> It's added, or postponed to when it should be added. I'm fine with whatever gets added or requested. I would recommend at least with issues like this is trying to use the templates in the future if you can please @GearBell. It does help.

**Comment by GearBell** [comment]
This is not a discussion its a tweak request. A simple tweak hotfix. Not an entire redesign. I know in the far future the UI will be reworked, but for now, Im requesting that the friend requests be shown at the top of the contacts list, not the bottom.

**Comment by Frooxius** [comment]
We could change the sorting for the existing UI, that's a relatively quick change, but some people prefer them lower and some people prefer them at the top, so it's a bit tricky issue, because no matter what we do, one side will be unhappy. We could perhaps do a poll to decide on that.

For anything more advanced than that the UI has to be redesigned properly, but it'll allow a lot more flexibility, personal customization, filtering and all that fancy stuff, but that'll take a bit more time!

**Comment by Psychpsyo** [comment]
If this gets changed, I'd like to request them to be placed below all online contacts in the list instead of above busy/afk people.
Please don't put them at the top, that'd be horrible with how many I keep around.

**Comment by Frooxius** [comment]
@Psychpsyo We can sort them in any way, but your request is unfortunately mutually exclusive with @GearBell's

Like I mentioned above, we can change it any way quickly, but it's going to be like that for everyone.

Anything customizable and configurable needs the UI to be reworked. Given the disagreement on how the sorting should be done, I'd probably just keep the status quo and invest the time into getting to the new UI instead.

**Comment by shiftyscales** [comment]
This issue is a duplicate, it was previously brought up, and shot down in #1864 because the new contacts UI will have some functionalities that will greatly assist the prevention of this issue.

> I think putting them at top would cause too many issues right now. I know some people are kinda keeping them there until they decide and this would require them scroll past them.
> 
> The new contacts UI is planned to have collapsible categories and filters, which should take care of this.

https://github.com/Neos-Metaverse/NeosPublic/issues/1864#issuecomment-806329344

I would also be in favor of putting them to the bottom just above offline users personally given the choice, but I'm probably an outlier in the number of unanswered contact requests I have because I try to get to know people before adding them back, and doing so would make the situation even worse for the situation GearBell had asked for this to resolve.

As Frooxius had noted, for some people like myself, this tweak would make things considerably worse, and I feel like overall, answering a contact request is an action done a lot less than inviting, messaging, and other functions you might want to do with online contacts.

I feel like there's a potential compromise that could be reached, but it would probably require a *lot* of work to implement, and that would be to ignore statuses, and solely sort the contacts list chronologically by last message sent/received, or for contacts with no message history, e.g. contact requests, place them at the top of the time of the request.

Discord does this in that your most recent conversations are *always* at the top of the list, regardless of status.

This would mean that contact requests are placed at the top as they are received, your most frequently contracted contacts are implicitly at the top, and there's a lot less to worry about with finding a user, because the list should always be relatively stable.

As noted though, doing this would probably take a considerably larger amount of work, assuming it could be done at all, @Frooxius ?

**Comment by Frooxius** [comment]
I could change it so they have the same level as Online users. But I'm not sure if they'll sort correctly at all times, since they don't have any active messages, so there won't be a timestamp to sort by. That'd require a potentially some more work to the old UI.

**Comment by GearBell** [comment]
> I could change it so they have the same level as Online users. But I'm not sure if they'll sort correctly at all times, since they don't have any active messages, so there won't be a timestamp to sort by. That'd require a potentially some more work to the old UI.<

No, that would most definatly become confusing. Either having them be on the top as "priority" or bottom as they are now would be better.

**Comment by Earthmark** [comment]
I feel like this topic is suggesting at two different things, it feels strange from my perspective to leave a contract request unanswered. Is the goal of it being unanswered that it's in a 'let's see if I really do like you' kinda state?

**Comment by shiftyscales** [comment]
> 
> 
> I feel like this topic is suggesting at two different things, it feels strange from my perspective to leave a contract request unanswered. Is the goal of it being unanswered that it's in a 'let's see if I really do like you' kinda state?

I'm in somewhat a unique position in terms of the volume of requests I receive compared to other community members- and it feels there is also a general culture wherein some users add each user in the session without a specific reason for it.

And yes, in my case, I usually leave them unanswered specifically because if I don't recognize the name, or don't have a particularly strong impression of them, I do not wish to return the invite, but if I find somebody I do wish to add because I enjoyed some aspect about them, and I see there's already a pending contact request, it allows me to add them right away/confirm the pending request.

The primary reason for doing this is because currently having a user as a contact has some heavy implications in terms of access, e.g. having access to any sessions you host set to contacts, or any session you are in set to contacts+- and also permissions for contacts being at a different level than guests/spectators.
